components/transaction/internal/services/command/create-bulk-transaction-operations-async.go

```
@@ -0,0 +1,762 @@
+
+// CreateBulkTransactionOperationsAsync processes multiple transaction messages in bulk.
+// This method:
+// 1. Unmarshals all payloads from the queue messages
+// 2. Processes balances for each transaction (required for hot balance)
+// 3. Extracts and sorts transactions/operations by ID (deadlock prevention)
+// 4. Performs bulk INSERTs for transactions and operations
+// 5. On bulk failure, falls back to individual processing if enabled
+//
+// The method returns a BulkResult with counts of attempted/inserted/ignored rows.
+// Individual message acknowledgment is the responsibility of the caller.
+func (uc *UseCase) CreateBulkTransactionOperationsAsync(
```

golangci-lint (2.11.3)

[error] cognitive complexity 32 of func `(*UseCase).CreateBulkTransactionOperationsAsync` is high (> 30)

(gocognit)